src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataControl.xaml.cs

```diff
@@ -1,85 +1,111 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.Globalization;
-using System.Threading;
+using System;
 using System.Windows;
 using System.Windows.Controls;
-using System.Windows.Documents;
-using NuGet.VisualStudio;
-using NuGet.VisualStudio.Telemetry;
+using Microsoft.VisualStudio.VCProjectEngine;
+using NuGet.PackageManagement.UI.ViewModels;
 
 namespace NuGet.PackageManagement.UI
 {
 
     /// <summary>
     /// Interaction logic for PackageMetadataControl.xaml
     /// </summary>
-    public partial class PackageMetadataControl : UserControl
+    public partial class PackageMetadataControl : UserControl, IDisposable
     {
+        private bool _disposed = false;
+        private TabItem SelectedTabItem
+        {
+            get
+            {
+                return tabsPackageDetails.SelectedItem as TabItem;
+            }
+            set
+            {
+                tabsPackageDetails.SelectedItem = value;
+            }
+        }
+
+        private ReadMePreviewViewModel _readMePreviewViewModel;
+
+        public PackageMetadataTab SelectedTab { get => (PackageMetadataTab)SelectedTabItem?.Tag; }
```

Comment:
Have you thought at all about Apex tests?

We've had product bugs in the past (and I'm sure we will again in the future) even when individual classes are unit tested, because integration bugs were not caught by those unit tests. We already have Apex tests (and many func tests) that create nupkg in a testing local source directory, we can modify it to add a readme in the nupkg. Then the Apex test can open PM UI (we already have tests that do this). This `SelectedTab` property is a getter only, which means maybe we need a way for tests to programatically switch to the readme tab, which would allow the test video recording to show us if the rendering is shown. We could have an assertion to check that the readme view model's readme markdown is not null to assert that it loaded the readme correctly from the nupkg. I'm not sure if there's anything else we could assert to check that the readme is actually being displayed.   Maybe check with the team that owns the markdown renderer to see if they provide any support for apex testing?

Reply:
I haven't! Is there a good overview for Apex tests somewhere?

Reply:
I created an issue for implementing APEX tests. I don't want to add more to this PR right now.